ci/python-publish.yml

```
@@ -12,28 +12,42 @@ on:
   release:
     types: [published]
 
-permissions:
-  contents: read
-
 jobs:
-  deploy:
+  release-build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
 
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.x"
+
+      - name: build release distributions
+        run: |
+          # NOTE: put your own distribution build steps here.
+          python -m pip install build
+          python -m build
```

*(this is probably controversial but)* the CI images come with pipx pre-provisioned so this could just use that:
```suggestion
          pipx run --spec build python -Im build
```

---

I'm -0 on this, for three reasons:

1. Fewer users know about `pipx`, so putting it here increases (very, very slightly) the cognitive complexity of the sample workflow. I recognize that this is a self-defeating argument, however, since this is the kind of thing needed to raise user awareness of `pipx` 🙂 
2. IMO `pipx run --spec` is a lot of indirection to introduce here: it requires users to think about `pipx`, about a venv for `build`, about a subprocess spawning `python` in that venv, etc. It's simpler for people like us, but I think this workflow also needs to cater to first-time publishers.
3. `pipx`'s availability on different runner images seems mixed/not subject to automated checks: https://github.com/actions/runner-images/issues/7590. Most users probably won't hit this since they'll be on `ubuntu-latest`, but it's another edge case we can dodge.

---

I agree.

---

Okay, I withdraw this idea.